mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -103,6 +103,13 @@ final class QuantileDiscretizer(override val uid: String)
 
 @Since("1.6.0")
 object QuantileDiscretizer extends DefaultParamsReadable[QuantileDiscretizer] with Logging {
+
+  /**
+   * Minimum number of samples required for finding splits, regardless of number of bins.  If
+   * the dataset has less rows than this value, the entire dataset column will be used.
+   */
+  val minSamplesRequired: Int = 10000
```

Comment:
Also, my original reason for asking about removing the hard coded value of 10K was because that value is the cause of the bug and so a regression test would need to know the value. 

I could have hard coded 10k in to my test.  However if a developer increased its value later, say to 100K, without increasing the hard coded test value as well, they could potentially render the test useless since it would always pass.